gr-uhd/apps/uhd_siggen_base.py

```diff
@@ -234,8 +246,48 @@ def set_amplitude(self, amplitude):
         else:
             return True # Waveform not yet set
         self.vprint("Set amplitude to:", amplitude)
+        self.update_gain_range()
         return True
 
+    def get_gain_or_power(self):
+        """
+        Depending on gain type, return either a power level or the current gain
+        """
+        if self.gain_type == self.GAIN_TYPE_GAIN:
+            return self.usrp.get_gain(self.channels[0])
+        # else:
```

Comment:
this line can be removed

Reply:
I often leave this in just as a reading aid, but I'm also happy to remove it.